### DISCUSSION TITLE
How to add `nnoremap` like mappings?

### DISCUSSION THREAD

**Discussion by rew1nter** [NvChad/NvChad#2105] [Q&A] [answered]
I had these `init.vim` lines that I'm now trying to convert to lua:
```vim
" Map Ctrl-Backspace to delete the previous word in insert mode.
noremap! <C-BS> <C-w>
noremap! <C-h> <C-w>

inoremap <C-w> <C-\><C-o>dB
inoremap <C-BS> <C-\><C-o>db


" Map Ctrl-Backspace to delete the previous word in normal mode.
nnoremap <C-h> db
```
I tried to put these in **custom/init.lua** file using this `vim.keymap.set` . But that didn't work, while something like this for example ```vim.keymap.set("i", "jj", "<Esc>")``` works fine. 
What am I doing wrong?

**Comment by siduck** [comment]
@rew1nter https://nvchad.com/docs/config/mappings , check the `opts` field there, so you just need `nnoremap = true` in it

**Reply by a-b** [reply]
Can you please elaborate on how to check `opts` field?  Is there any documentation? Example has only mention `Remember that the mappings must have a vim mode: n (for normal), v (for visual), i (for insert) and so on.` which is not helpful without relevant example.

**Reply by siduck** [reply]
![image](https://github.com/NvChad/NvChad/assets/59060246/fa3088de-a1d1-4dfe-b5bb-f1955ea6396f)

@a-b nvchad mappings use vim.keymap.set under the hood, so you see that opts table? thats used in nvchad too

`["keys"] = {"action", "description", opts = { THISSSSSSS }},`

**Reply by a-b** [reply]
Thanks this is helpful. I'll make PR with additional example for the visual mode.

**Comment by rew1nter** [comment]
```lua
["<C-h>"] = {"db", "Ctrl+bak delete word in normal mode", opts={nnoremap= true}},
```
Setting this throws an error but works as expected if I remove the `opts` :/

How do I set the `inoremap` commands in this case?

**Reply by siduck** [reply, marked as answer]
@rew1nter sorry it was noremap! this should work

![image](https://github.com/NvChad/NvChad/assets/59060246/f204c781-aceb-429e-918a-d889e3d192f3)

**Comment by rew1nter** [comment]
Thanks!
Did I just manage to get someone in the CTRL+bak team XD

**Reply by siduck** [reply]
haha no, i tested it only for you :)
